test/pytorch_test_base.py

```diff
@@ -143,9 +143,10 @@
     'test_randn_xla_float64',  # xla doesn't support manual_seed, as_stride
     'test_rand_xla_float32',  # xla doesn't support manual_seed, as_stride
     'test_rand_xla_float64',  # xla doesn't support manual_seed, as_stride
-    'test_normal_xla_float64',  # AssertionError: 0.22364577306378963 not less than or equal to 0.2
+    'test_normal',  # AssertionError: 0.22364577306378963 not less than or equal to 0.2
     'test_uniform_from_to',  # Checks for error strings.
     'test_bucketization', # test expect an non-contiguous tensor but xla permute op will return a contigous tensor
+    'test_index_fill_xla', # half support
```

Comment:
Can we just disable `half` type?

Reply:
The test looks like
```
    def test_index_fill(self, device):                                                                                                                                                                                             
        for dt in torch.testing.get_all_dtypes():                                                                                                                                                                                  
            if dt == torch.half or dt == torch.bfloat16:                                                                                                                                                                           
                continue                                                                                                                                                                                                           
                                                                                                                                                                                                                                   
            x = torch.tensor([[1, 2], [4, 5]], dtype=dt, device=device)                                                                                                                                                            
            index = torch.tensor([0], device=device)                                                                                                                                                                               
            x.index_fill_(1, index, 0)                                                                                                                                                                                             
            self.assertEqual(x, torch.tensor([[0, 2], [0, 5]], dtype=dt, device=device)) 
```
we will fail when we try to allocate a tensor with `half` type. The only way on the top of my head is to overwrite `torch.testing.get_all_dtypes` to not return `half` but that might be a bit hacky.. Need to take a look at how it is being implemented.

Reply:
It would be better if test is written like
``
    @dtypes(torch.double, torch.half. torch.float32 ..........)
    def test_index_fill(self, device):                                                                                                                                                                                             
```
then we can disable the `half` variant of the test

Reply:
Ohh, I see. Does not do normal test-name-expansion.

Reply:
The correct fix should be that the testing class has an API to return the supported types.